pkg/plugins/golang/v3/commons.go

```diff
@@ -145,7 +145,7 @@ manifests: controller-gen`
 		}
 
 		if err := util.ReplaceInFile("go.mod",
-			"k8s.io/klog/v2 v2.30.0",
+			"k8s.io/klog/v2 v2.60.1",
 			"k8s.io/klog/v2 v2.9.0"); err != nil {
 			log.Warnf("unable to update the go.mod with k8s.io/klog/v2 v2.9.0: %s", err)
 		}
```

Comment:
After this line we should add:
```go
// Due to some indirect dependency changes with a bump in k8s packages from 0.23.x --> 0.24.x we need to
// clear out all indirect dependencies before we run `go mod tidy` so that the above changes get resolved correctly
if err := util.ReplaceRegexInFile("go.mod", `(require \(\n(\t.* \/\/ indirect\n)+\))`, ""); err != nil {
	log.Warnf("unable to update the go.mod indirect dependencies: %s", err)
}
```

The reasoning for this change can be seen here: https://github.com/kubernetes-sigs/kubebuilder/pull/2679#issuecomment-1146263855